### PR TITLE
clock: fixes for turbo cheat

### LINKED ISSUE #1341: Animated textures issue when using different turbo speeds - 60 FPS
I noticed these 2 things can happen:

- Animated textures become static when using negative turbo speeds.
- Animated textures speed remain unchanged when using positive turbo speeds.

This may take several attempts to reproduce so keep toggling between 0 and -1 until the animation stops.  The issue is not present in 30 FPS mode.

**A good spot to test this issue in Lost Valley:**
Screenshot

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [X] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

Resolves #1341

### CODE REVIEW

**Review by lahm86 — COMMENTED** [review]
This looks good, other than the pickups, which I think @aredfan has mentioned already.
- `-2` 3D pickups don't appear (all other values are fine)
- `-2` or `-1` static pickups stay on screen permanently (all others fine)

20240425_203444_Lost_Valley

Interestingly, if you go in and out of the inventory twice, the pickup display is then cleared.

**Comment by rr-** [comment]
I've reworked the fix; needs lots of testing around everything that uses clock…

I don't like the current clock API, if you have ideas how to reduce its size I'm all ears.

Scenario matrix:

- inventory spinout and cycling animation speed
- inventory item animations (medpack opening, passport page flips etc)
- inventory stats timer
- text flash rate (binding conflicts)
- health bar flash rate
- 3d pickups animation rate
- animated textures animation rate
- fades animation rate
- underwater wibble effect speed
- picture display time (loading screens and credits)
- title screen demo delay

×

- 60 FPS
- 30 FPS

×

- VSync ON
- Vsync OFF
(Testing this is crucial, as VSync effectively restricts the FPS to match the screen refresh rate; without it, turbo +2 renders the game at ~120 FPS. This FPS cap makes calculations extra tricky, but I'm hoping I got it right this time.)

×

- Turbo speed -2
- Turbo speed -1
- Turbo speed 0
- Turbo speed +1
- Turbo speed +2

Vectors I intentionally did not fix because I was too drained to work on it any longer + I don't think they have much impact: key reset button hold time, target lock buttons hold time.

**Comment by rr-** [comment]
@aredfan if you're up to the task ↑

https://nebula.wind.garden/serve/pub/TR1X-4.0.3-38-g02d390e-Windows.zip

**Comment by rr-** [comment]
Added `/vsync on` and `/vsync off` command for a little bit easier testing.

https://nebula.wind.garden/serve/pub/TR1X-4.0.3-39-g6794f1d-Windows.zip

File: src/game/console_cmd.c [hunk 1]
```diff
@@ -236,6 +237,27 @@ static bool Console_Cmd_Speed(const char *const args)
     return false;
 }
 
+static bool Console_Cmd_VSync(const char *const args)
+{
+    if (String_Equivalent(args, "off")) {
+        g_Config.rendering.enable_vsync = false;
+        Config_Write();
+        Output_ApplyRenderSettings();
```
Comment (walkawayy) [inline src/game/console_cmd.c:246]: Need to include `output.h` because you use `Output_ApplyRenderSettings`

**Review by walkawayy — CHANGES_REQUESTED** [review]

**Comment by walkawayy** [comment]
All your situations seemed good to me except for the warning that `output.h` isn't included.

File: src/game/console_cmd.c (hunk 1 above)
Comment (rr-) [reply src/game/console_cmd.c:246]: Thank you for spotting this; I have added the include file.

**Comment by aredfan** [comment]
I been testing all the scenarios and I think this PR is good for the most part. I've only found a few minor issues.

**This only happens with:**

- 60 FPS
- VSync ON
- Turbo speeds +1 and +2

**Issues:**

- The inventory items hangs on the last frame a little while longer when the ring is closing.
- Loading screens and credits fades isn't sped up.
- Fades have trouble keeping up when opening/closing the inventory.

**Comment by rr-** [comment]
@aredfan thank you for testing. I believe I have fixed the issues mentioned above. At the same I managed to land a big cleanup of the clock API – it's much more readable ATM. Unfortunately this change can impact things that did work before, so it'd be good to have a full scenario check.

https://nebula.wind.garden/serve/pub/TR1X-4.0.3-40-g491e9f1-Windows.zip

**Comment by aredfan** [comment]
@rr- You're very welcome.  👍 

I've re-tested all the scenarios and most things LGTM.  I ran into a few new minor issues which only happen in 30 FPS mode regardless of whether VSync is enabled or disabled.

- The inventory items hangs on the last frame a little while when the ring is closing - all turbo speeds
- Fades struggle to keep up on quickload - all turbo speeds
- Fades struggle to keep up with the inventory opening - all turbo speeds

**Comment by rr-** [comment]
@aredfan I believe this should fix the fades for 30 FPS:
https://nebula.wind.garden/serve/pub/TR1X-4.0.3-41-g0771edb-Windows.zip
Maybe even inventory closing. The changes are isolated to fades alone, so no need to retest the other things.

**Comment by aredfan** [comment]
@rr- Yep your last commit did the trick.  All outstanding issues are resolved including the inventory closing.  Thank you again. 👍

**Review by walkawayy — APPROVED** [review]
Nice thanks for the testing help. Looks good to me.

**Review by lahm86 — APPROVED** [review]

**Pull request merged**